controllers/replace_failed_pods.go

```diff
@@ -52,29 +53,50 @@ func chooseNewRemovals(cluster *fdbtypes.FoundationDBCluster) bool {
 		return false
 	}
 
+	// The maximum number of removals will be the defined number in the cluster spec
+	// minus all currently ongoing removals e.g. process groups marked fro removal but
+	// not fully excluded.
+	removalCnt := 0
 	for _, processGroupStatus := range cluster.Status.ProcessGroups {
 		if processGroupStatus.Remove && !processGroupStatus.Excluded {
 			// If we already have a removal in-flight, we should not try
 			// replacing more failed pods.
-			return false
+			removalCnt++
 		}
 	}
+	maxReplacements := cluster.GetMaxConcurrentReplacements() - removalCnt
 
+	hasReplacement := false
 	for _, processGroupStatus := range cluster.Status.ProcessGroups {
+		if maxReplacements == 0 {
```

Comment:
```suggestion
		if maxReplacements <= 0 {
```

There are cases where we have a negative number --> Add a test case for that.